core/src/snapshot/postprocessers/postprocess.rs

```diff
@@ -0,0 +1,198 @@
+use heimdall_common::utils::strings::{
+    classify_token, find_balanced_encapsulator, tokenize, TokenType,
+};
+
+pub fn remove_double_negation(line: &str) -> String {
+    let mut cleaned = line.to_owned();
+
+    while cleaned.contains("((!((!((") {
+        // find the indices of the subject
+        let subject_indices = cleaned.find("((!((!((").unwrap();
+        let subject = cleaned[subject_indices..].to_string();
+
+        // get the indices of the subject's first negation encapsulator
+        let first_subject_indices = find_balanced_encapsulator(&subject, ('(', ')'));
+        if first_subject_indices.2 {
+            // the subject to search is now the subject without the first negation encapsulator
+            let second_subject = subject[first_subject_indices.0 + 1..].to_string();
+
+            // get the indices of the subject's second negation encapsulator
+            let second_subject_indices = find_balanced_encapsulator(&second_subject, ('(', ')'));
+            if second_subject_indices.2 {
+                // the subject is now the subject without the first and second negation encapsulators
+                let subject = second_subject
+                    [second_subject_indices.0 + 1..second_subject_indices.1 - 1]
+                    .to_string();
+
+                // remove the double negation
+                cleaned.replace_range(
+                    subject_indices
+                        ..subject_indices + first_subject_indices.0 + 2 + second_subject_indices.1,
+                    &subject,
+                );
+            }
+        }
+    }
+
+    cleaned
+}
+
+pub fn simplify_parentheses(line: &str, paren_index: usize) -> String {
+    // helper function to determine if parentheses are necessary
+    fn are_parentheses_unnecessary(expression: &str) -> bool {
```

Comment:
nice! since this is the same as the helper in the `decompile` module, can you add a TODO to move to `heimdall-common`?